### ISSUE TITLE
Doesn't hot reload required modules

### ISSUE THREAD

**Issue by mxrlkn** [iiroj/lambda-dev#6]
Just found that it doesn't hot reload code that has been required.
`const utils = require("../modules/utils.js");`.

editing utils.js does output `λ-dev Built`, but the new edit doesn't run when you hit the handler.

I've looked a bit through the code, and I guess this doesn't account for required sources:

https://github.com/iiroj/lambda-dev/blob/9dbdf5de70fb4706daf76afee096acf525aef1bf/src/serve.ts#L48-L50

I'm a bit unsure how to fix this it without being 100% on webpack internals.

**Comment by dcsaszar** [comment]
I'm observing the same issue (λ-dev built, but the executed code is outdated), and I'm also unsure how to fix it.
Maybe there is a quickfix that re-runs as much as possible of the initial compile process on a change?

**Comment by mxrlkn** [comment]
@iiroj any ideas?

**Comment by iiroj** [comment]
Thanks for the feedback. One way to solve it would definitely be to re-build everything on changes. Another idea would be to empty node's require caches.

I will see if I can find a way to do this nicely. In the meantime, sorry for the inconvenience.

**Comment by iiroj** [comment]
Just FYI, I did manage to repeat this issue.

**Comment by iiroj** [comment]
I think I managed to fix this issue with https://github.com/iiroj/lambda-dev/pull/7

The way it works is:
0. Hook into Webpack compilation's `watchRun` hook, which runs in watch mode when files are changed
0. get a list of changed files from the compilation in the hook
0. if node's require cache has cached the changed file, delete it from cache
0. this will result in the file being required again

I tested this by creating a new file outside the `test/fixtures` directory that exports a string, and then editing the `hello-world.js` test file to send the contents of this file. When I edited the new file, `lambda-dev` reported a build in the console, and after reloading my browser the test string matched the new version.

What do you think?

**Comment by iiroj** [comment]
I released this PR as a pre-release `v1.8.0-0`. You can test it out by installing:

```bash
npm i --save-dev lambda-dev@1.8.0-0
```

**Comment by dcsaszar** [comment]
Confirmed.

**Comment by iiroj** [comment]
I released the fix in `v1.8.0`:

```bash
npm i --save-dev lambda-dev@^1.8.0
```

I'll close this issue. Thanks for the report!
